test/e2e/apps/replica_set.go

```diff
@@ -524,3 +535,135 @@ func testRSLifeCycle(f *framework.Framework) {
+	framework.ExpectNoError(err, "Failed to fetch the status of replicaset %s in namespace %s", rsName, ns)
+	rsStatusBytes, err := json.Marshal(rsStatusUnstructured)
+	framework.ExpectNoError(err, "Failed to marshal unstructured response. %v", err)
+
+	var rsStatus appsv1.ReplicaSet
+	err = json.Unmarshal(rsStatusBytes, &rsStatus)
+	framework.ExpectNoError(err, "Failed to unmarshal JSON bytes to a replicaset object type")
+	framework.Logf("Replicaset %s has Conditions: %v", rsName, rsStatus.Status.Conditions)
+
+	ginkgo.By("updating the Replicaset Status")
+	var statusToUpdate, updatedStatus *appsv1.ReplicaSet
+
+	err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		statusToUpdate, err = rsClient.Get(context.TODO(), rsName, metav1.GetOptions{})
+		framework.ExpectNoError(err, "Unable to retrieve replicaset %s", rsName)
+
+		statusToUpdate.Status.Conditions = append(statusToUpdate.Status.Conditions, appsv1.ReplicaSetCondition{
+			Type:    "StatusUpdate",
+			Status:  "True",
+			Reason:  "E2E",
+			Message: "Set from e2e test",
+		})
+
+		updatedStatus, err = rsClient.UpdateStatus(context.TODO(), statusToUpdate, metav1.UpdateOptions{})
+		return err
+	})
+	framework.ExpectNoError(err, "Failed to update status. %v", err)
+	framework.Logf("updatedStatus.Conditions: %#v", updatedStatus.Status.Conditions)
+
+	ginkgo.By("watching for the ReplicaSet status to be updated")
+	ctx, cancel := context.WithTimeout(context.Background(), rsRetryTimeout)
+	defer cancel()
+	_, err = watchtools.Until(ctx, rsList.ResourceVersion, w, func(event watch.Event) (bool, error) {
+		if rs, ok := event.Object.(*appsv1.ReplicaSet); ok {
+			found := rs.ObjectMeta.Name == testReplicaSet.ObjectMeta.Name &&
+				rs.ObjectMeta.Namespace == testReplicaSet.ObjectMeta.Namespace &&
+				rs.ObjectMeta.Labels["name"] == testReplicaSet.ObjectMeta.Labels["name"] &&
+				rs.ObjectMeta.Labels["pod"] == testReplicaSet.ObjectMeta.Labels["pod"]
+			if !found {
+				framework.Logf("Observed replicaset %v in namespace %v with annotations: %v & Conditions: %v", rs.ObjectMeta.Name, rs.ObjectMeta.Namespace, rs.Annotations, rs.Status.Conditions)
+				return false, nil
+			}
+			for _, cond := range rs.Status.Conditions {
+				if cond.Type == "StatusUpdate" &&
+					cond.Reason == "E2E" &&
+					cond.Message == "Set from e2e test" {
+					framework.Logf("Found replicaset %v in namespace %v with labels: %v annotations: %v & Conditions: %v", rs.ObjectMeta.Name, rs.ObjectMeta.Namespace, rs.ObjectMeta.Labels, rs.Annotations, rs.Status.Conditions)
+					return found, nil
+				}
+				framework.Logf("Observed replicaset %v in namespace %v with annotations: %v & Conditions: %v", rs.ObjectMeta.Name, rs.ObjectMeta.Namespace, rs.Annotations, rs.Status.Conditions)
+			}
+		}
+		object := strings.Split(fmt.Sprintf("%v", event.Object), "{")[0]
+		framework.Logf("Observed %v event: %+v", object, event.Type)
+		return false, nil
```

Comment:
@ravisantoshgudimetla Thanks for the earlier review. I've updated this section as there was a potential bug where I had an extra `return`. Also, line 627-8 includes the object for the event now.

Can I get another lgtm? TIA